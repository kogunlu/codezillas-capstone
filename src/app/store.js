import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import bookingReducer from '../features/booking/bookingSlice';
import userReducer from '../features/user/userSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    booking: bookingReducer,
    user: userReducer,
  },
});
