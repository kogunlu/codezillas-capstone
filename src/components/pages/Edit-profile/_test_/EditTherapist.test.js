import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../app/store';
import EditTherapist from '../EditTherapist';

it('Renders correctly for EditUser Component', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <EditTherapist />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
