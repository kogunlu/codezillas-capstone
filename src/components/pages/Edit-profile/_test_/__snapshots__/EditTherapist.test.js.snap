// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly for EditUser Component 1`] = `
<div
  className="flex flex-col justify-start items-center h-full my-5 gap-5"
>
  <div
    className="w-5/6"
  >
    <p
      className="text-center text-red-500 font-semibold text-lg"
    >
      Please fill al the fields with correct and valid details to complete your profile.
    </p>
  </div>
  <div
    className="w-5/6 h-full flex flex-col justify-center items-center gap-5 lg:flex-row lg:justify-between lg:items-start lg:gap-0"
  >
    <div
      className="w-2/6 h-full"
    >
      <span
        className="flex justify-center "
      >
        <img
          alt="userAvatar"
          className="h-60 xl:h-80 -z-10 "
          src="user.jpg"
        />
      </span>
      <button
        className="z-20 -mt-10 w-full flex justify-center"
        type="button"
      >
        <img
          alt="editUser"
          className="h-12 border-black border-2 rounded-full"
          src="edit-user.jpg"
        />
      </button>
    </div>
    <div
      className="w-full lg:w-3/6 "
    >
      <div>
        <p
          className="text-4xl font-bold px-5 text-center lg:text-start"
        >
          PROFILE INFO
        </p>
        <form
          className="w-full h-full py-10 flex flex-col justify-between gap-5 items-center px-2 md:px-5"
          onSubmit={[Function]}
        >
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Full Name
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="fullName"
                name="fullName"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-56 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Bio
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col relative"
            >
              <textarea
                className="border shadow-md w-full h-44 md:h-48 rounded-md pl-2 pt-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-start"
                id="bio"
                name="bio"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Birth Date
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base text-white focus:text-black"
                id="birthdate"
                max="2022-12-31"
                min="1900-01-01"
                name="birthdate"
                onBlur={[Function]}
                onChange={[Function]}
                type="date"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Email
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full h-20 flex justify-between"
          >
            <span
              className="w-4/12 h-12 md:h-16 flex justify-start items-center"
            >
              <p
                className="text-xl font-semibold"
              >
                Phone
              </p>
            </span>
            <div
              className="w-8/12 flex flex-col"
            >
              <input
                className="border shadow-md w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                id="phone"
                maxLength="16"
                name="phone"
                onBlur={[Function]}
                onChange={[Function]}
                pattern="[0-9]{2} [0-9]{3} [0-9]{3} [0-9]{2} [0-9]{2}"
                placeholder="90 555 999 99 99"
                type="tel"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full flex flex-col flex-wrap items-center gap-5 md:flex-row md:justify-between"
          >
            <input
              className="py-2 px-2 w-10/12 md:w-3/12 lg:w-5/12 xl:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
              type="submit"
              value="SAVE CHANGES"
            />
            <input
              className="py-2 px-2 w-10/12 md:w-3/12 lg:w-5/12 xl:w-3/12 hover:bg-red-600 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
              onClick={[Function]}
              type="button"
              value="DELETE ACCOUNT"
            />
            <input
              className="py-2 px-2 w-10/12 md:w-3/12 lg:w-5/12 xl:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-semibold rounded text-sm md:text-md shadow-lg cursor-pointer"
              onClick={[Function]}
              type="button"
              value="CANCEL"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
